### PR TITLE
 #559044: [SXA] fixed rendering dynamic placeholder

### PR DESCRIPTION
**Pull request by matkovskyi**
## Description / Motivation
Rewrote code which deleted temporary dynamic placeholder.
Problem was in hardcoded prefix of name dynamic placeholder.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by StanislavStativa — APPROVED** [review]

File: packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx [hunk 1]
```diff
@@ -121,10 +118,17 @@ export class PlaceholderCommon<T extends PlaceholderProps> extends React.Compone
...
-    if (rendering && rendering.placeholders && rendering.placeholders[PREFIX_PLACEHOLDER]) {
-      rendering.placeholders[name] = rendering.placeholders[PREFIX_PLACEHOLDER];
-      delete rendering.placeholders[PREFIX_PLACEHOLDER];
+    from backend side we get common name of placeholder is called 'nameOfContainer-{*}' where '{*}' marker for replacing **/
+    const isExistsDynamicPlaceholders =
+      rendering?.placeholders &&
+      Object.keys(rendering.placeholders).find((placeholder) => placeholder.indexOf('{*}') !== -1);
+    if (isExistsDynamicPlaceholders) {
+      Object.keys(rendering.placeholders).forEach((placeholder) => {
+        if (placeholder.indexOf('container-{*}') !== -1) {
```
Comment (ambrauer) [inline packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx:127]: I'm confused about the expected placeholder name format. In the `isExistsDynamicPlaceholders` we are checking for `{*}` but here we are using `container-{*}`. Is "container-" always present or not? If so, we could keep the constant, but rename it as postfix e.g. `DYNAMIC_PLACEHOLDER_POSTFIX = 'container-{*}';`

**Review by ambrauer — COMMENTED** [review]
See some questions/comments

File: packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx (hunk 1 above)
Comment (matkovskyi) [reply packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx]: @ambrauer My bad. This `container-{*}` was for testing and old code. I've fixed it

File: packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx [hunk 2]
```diff
@@ -121,10 +118,17 @@ export class PlaceholderCommon<T extends PlaceholderProps> extends React.Compone
   ) {
     let result;
     /** [SXA] it needs for deleting dynamics placeholder when we set him number(props.name) of container.
-    from backend side we get common name of placeholder is called 'container-{*}' where '{*}' marker for replacing **/
-    if (rendering && rendering.placeholders && rendering.placeholders[PREFIX_PLACEHOLDER]) {
-      rendering.placeholders[name] = rendering.placeholders[PREFIX_PLACEHOLDER];
-      delete rendering.placeholders[PREFIX_PLACEHOLDER];
+    from backend side we get common name of placeholder is called 'nameOfContainer-{*}' where '{*}' marker for replacing **/
+    const isExistsDynamicPlaceholders =
```
[the remark this answers is not in the document]
Comment (matkovskyi) [reply packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx]: @ambrauer Yes, you're right it was redundant

**Review by ambrauer — APPROVED** [review]
Looks good!

**Pull request merged**